apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/SecurityTagManager.java

```diff
@@ -0,0 +1,144 @@
+
+    Query query = entityManager.createQuery(sql);
+    query.setParameter("claim", claimId);
+
+    @SuppressWarnings("unchecked")
+    List<String> resultList = query.getResultList();
+
+    return new HashSet<>(resultList);
+  }
+
+  /**
+   * Determines the security level based on the collected tags.
+   *
+   * @param claimId value of claimId
+   * @param tagClass value of tagClass
+   * @return SecurityLevel
+   */
+  public List<Coding> getClaimSecurityLevel(String claimId, Class<?> tagClass) {
+
+    if (samhsaV2Enabled) {
+      // Query tags associated with the claim
+      List<String> securityTags = queryTagsForClaim(claimId, tagClass).stream().toList();
+
+      List<Coding> securityTagCoding = new ArrayList<>();
+
+      // If no security tags are found, directly add the default "Normal" tag
+      if (securityTags.isEmpty()) {
+        Coding coding = new Coding();
+        coding
+            .setSystem(TransformerConstants.SAMHSA_CONFIDENTIALITY_CODE_SYSTEM_URL)
+            .setCode("N")
+            .setDisplay("Normal");
+        securityTagCoding.add(coding);
+      } else {
+        // Check for each tag and set corresponding code and display
+        for (String securityTag : securityTags) {
+          Coding coding = new Coding();
+          // Convert the securityTag string to the TagCode enum
+          TagCode tagCode = TagCode.fromString(securityTag);
+          // Check each security tag and apply corresponding values
+          if (tagCode != null) {
+            switch (tagCode) {
+              case R:
+                coding
+                    .setSystem(TransformerConstants.SAMHSA_CONFIDENTIALITY_CODE_SYSTEM_URL)
+                    .setCode(TagCode.R.toString())
+                    .setDisplay(TagCode.R.getDisplayName());
+                break;
+              case _42CFRPart2:
+                coding
+                    .setSystem(TransformerConstants.SAMHSA_ACT_CODE_SYSTEM_URL)
+                    .setCode(TagCode._42CFRPart2.toString())
+                    .setDisplay(TagCode._42CFRPart2.getDisplayName());
+                break;
+            }
+          }
+          securityTagCoding.add(coding);
+        }
+      }
+      return securityTagCoding;
+    }
+    return new ArrayList<>();
+  }
+
+  /**
+   * Determines the security level based on the collected tags.
+   *
+   * @param claimId value of claimId
+   * @param tagClass value of tagClass
+   * @return SecurityLevel
+   */
+  public List<org.hl7.fhir.dstu3.model.Coding> getClaimSecurityLevelDstu3(
+      String claimId, Class<?> tagClass) {
+
+    List<org.hl7.fhir.dstu3.model.Coding> securityTagCoding = new ArrayList<>();
+    List<Coding> coding = getClaimSecurityLevel(claimId, tagClass);
+    for (Coding code : coding) {
+      org.hl7.fhir.dstu3.model.Coding securityTag = new org.hl7.fhir.dstu3.model.Coding();
+      securityTag
+          .setSystem("http://terminology.hl7.org/CodeSystem/v3-Confidentiality")
```

Comment:
This should use the constant
```suggestion
          .setSystem(TransformerConstants.SAMHSA_CONFIDENTIALITY_CODE_SYSTEM_URL)
```